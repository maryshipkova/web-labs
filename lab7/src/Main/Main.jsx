import React from 'react';
import {Weather} from "../Weather/Weather";
import {Bookmarks} from "../Bookmarks/Bookmarks";
import {useDispatch, useSelector} from "react-redux";
import './Main.scss'
import {remove} from "../store/actions";

export const Main = (props) => {
    const [coordinates, updateCoordinates] = React.useState();
    const [loading, isLoading] = React.useState(false);
    const cities = useSelector(state => state.cities);
    const dispatch = useDispatch();
    const onButtonClick = () => {
        isLoading(true);

        navigator.geolocation.getCurrentPosition(function (position) {
                const latitude = position.coords.latitude;
                const longitude = position.coords.longitude;
                updateCoordinates({lat: latitude, lon: longitude});
                isLoading(false);
            },
            function (err) {
                console.log(err);
                updateCoordinates({lat: '59', lon: '30'});
                isLoading(false);
            }, {
                maximumAge:1
            });

    };

    const onRemove = (city) => {
        remove(dispatch, city);
    };

    return (
        <main className={'Main'}>
            <div className="Main-Search">
                <h3>Погода здесь</h3>
                <button className="Main-Button" onClick={onButtonClick}>Обновить геолокацию</button>
            </div>
            {
                props.loading || loading ? <div>loading</div> :
                    <Weather main coordinates={coordinates}/>
            }
            <div className={'Main-Bookmarks'}>
                <Bookmarks/>
                {
                    cities && cities.map(city => <Weather id={city} city={city} key={city} onRemove={onRemove}/>)
                }
            </div>
        </main>
    )
};
