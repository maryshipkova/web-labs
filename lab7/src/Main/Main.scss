@import '../constants.scss';

.Main {
  display: flex;
  flex-direction: column;

  background-color: white;
  border-radius: 18px;
  min-width: 50%;
  min-height: 200px;
  padding: 10px 20px;

  &-Button{
    background-color:$color-gray;
    padding: 10px 15px;
    border-radius: 100px;
    border: none;
    font-size: 16px;
    font-weight: bold;
  }
}
