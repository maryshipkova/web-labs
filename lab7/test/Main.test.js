import React from 'react';
import configureStore from "redux-mock-store";
import thunk from "redux-thunk";
import renderer from "react-test-renderer";
import {Provider} from "react-redux";
import {Main} from "../src/Main/Main";

describe('Main', () =>{
    beforeAll(()=>{
        console.error = jest.fn();
        window.fetch = jest.fn(()=>Promise.resolve());
    });

    it('plain', () => {
        const initialState={cities: ["Moscow","1234","12345"]};
        const mockStore = configureStore([thunk]);
        const store = mockStore(initialState);

        let tree = renderer
            .create(
                <Provider store={store}>
                    <Main />
                </Provider>);
        expect(tree.toJSON()).toMatchSnapshot();
    });

    it('loading', () => {
        const initialState={cities:[]};
        const mockStore = configureStore([thunk]);
        const store = mockStore(initialState);

        let tree;
        renderer.act(() => {
            tree = renderer
                .create(
                    <Provider store={store}>
                        <Main loading/>
                    </Provider>);
        });

        expect(tree.toJSON()).toMatchSnapshot();
    });
});





// const initialState={cities: ["Moscow","1234","12345"]};
// const mockStore = configureStore([thunk]);
// const store = mockStore(initialState);
// window.fetch = jest.fn(()=>Promise.resolve());
// const tree = shallow(
//             <Provider store={store}>
//                 <Main loading/>
//             </Provider>)
// expect(renderToJson(tree)).toMatchSnapshot();
