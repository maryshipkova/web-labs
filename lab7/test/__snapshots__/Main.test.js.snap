// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main loading 1`] = `
<main
  className="Main"
>
  <div
    className="Main-Search"
  >
    <h3>
      Погода здесь
    </h3>
    <button
      className="Main-Button"
      onClick={[Function]}
    >
      Обновить геолокацию
    </button>
  </div>
  <div>
    loading
  </div>
  <div
    className="Main-Bookmarks"
  >
    <main
      className="Bookmarks"
    >
      <div
        className="Bookmarks-Row"
      >
        <h3
          className="Bookmarks-Title"
        >
          Избранное
        </h3>
        <form
          className="Bookmarks-Add"
          onSubmit={[Function]}
        >
          <input
            className="Bookmarks-Input"
            onChange={[Function]}
            placeholder="Добавить новый город"
            type="text"
          />
          <input
            className="Bookmarks-AddButton Weather-Button"
            type="submit"
            value="+"
          />
        </form>
      </div>
    </main>
  </div>
</main>
`;

exports[`Main plain 1`] = `
<main
  className="Main"
>
  <div
    className="Main-Search"
  >
    <h3>
      Погода здесь
    </h3>
    <button
      className="Main-Button"
      onClick={[Function]}
    >
      Обновить геолокацию
    </button>
  </div>
  <div
    className="Weather Weather_main"
  />
  <div
    className="Main-Bookmarks"
  >
    <main
      className="Bookmarks"
    >
      <div
        className="Bookmarks-Row"
      >
        <h3
          className="Bookmarks-Title"
        >
          Избранное
        </h3>
        <form
          className="Bookmarks-Add"
          onSubmit={[Function]}
        >
          <input
            className="Bookmarks-Input"
            onChange={[Function]}
            placeholder="Добавить новый город"
            type="text"
          />
          <input
            className="Bookmarks-AddButton Weather-Button"
            type="submit"
            value="+"
          />
        </form>
      </div>
    </main>
    <div
      className="Weather"
    />
    <div
      className="Weather"
    />
    <div
      className="Weather"
    />
  </div>
</main>
`;
